src/metabase/models/action.clj

```diff
@@ -282,7 +289,20 @@
                                 (map #(dissoc % ::pk? ::field-id)))]
           (cond-> (assoc action :database_id (model-id->db-id (:model_id action)))
             (seq implicit-params)
-            (assoc :parameters implicit-params)))
+            (-> (assoc :parameters implicit-params)
+                (update-in [:visualization_settings :fields]
+                           (fn [fields]
+                             (let [param-ids (map :id implicit-params)]
+                               (let [fields (->> (or fields {})
+                                                 ;; remove entries that don't match params
+                                                 (m/filter-keys (set param-ids)))]
+                                 ;; add default entries for params that don't have an entry
```

Comment:
why having an existing entry is important here?
I'm trying to understand why you doing this instead of merge first with a default value like this

```clojure
(let [param-ids (map :id implicit-params)
      fields    (->> (or fields {})
                     ;; remove entries that don't match params
                     (m/filter-keys (set param-ids)))]
  (reduce (fn [acc param-id]
            (update acc param-id #(merge {:id param-id, :hidden false} %)))
          fields
          param-ids))
```